drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -311,6 +380,73 @@ void parseURDF(RigidBodySystem *sys, XMLDocument *xml_doc)
   parseRobot(sys, node);
 }
 
+void parseSDFJoint(RigidBodySystem *sys, string model_name, XMLElement* node, PoseMap& pose_map)
+{
+  // todo: parse joint sensors
+}
+
+void parseSDFLink(RigidBodySystem *sys, string model_name, XMLElement* node, PoseMap& pose_map)
+{
+  const char* attr = node->Attribute("name");
+  if (!attr) throw runtime_error("ERROR: link tag is missing name attribute");
+  string link_name(attr);
+
+  auto body = sys->getRigidBodyTree()->findLink(link_name, model_name);
+
+  Isometry3d transform_link_to_model = Isometry3d::Identity();
+  XMLElement* pose = node->FirstChildElement("pose");
+  if (pose) {
+    poseValueToTransform(pose, pose_map, transform_link_to_model);
+    pose_map.insert(std::pair<string,Isometry3d>(body->linkname, transform_link_to_model));
+  }
+
+  for (XMLElement *elnode = node->FirstChildElement("sensor"); elnode; elnode = elnode->NextSiblingElement("sensor")) {
+    attr = elnode->Attribute("name");
+    if (!attr) throw runtime_error("ERROR: sensor tag is missing name attribute");
+    string sensor_name(attr);
+
+    attr = elnode->Attribute("type");
+    if (!attr) throw runtime_error("ERROR: sensor tag is missing type attribute");
+    string type(attr);
+
+    Isometry3d transform_sensor_to_model = transform_link_to_model;
+    XMLElement* pose = elnode->FirstChildElement("pose");
+    if (pose) {
+      poseValueToTransform(pose, pose_map, transform_sensor_to_model, transform_link_to_model);
+    }
+
+    if (type.compare("ray")==0) {
+      auto frame = allocate_shared<RigidBodyFrame>(Eigen::aligned_allocator<RigidBodyFrame>(),sensor_name,body,transform_link_to_model.inverse()*transform_sensor_to_model);
```

Comment:
I think the fact that it is an Isometry3d means that Eigen knows how to optimize the inverse.  (http://eigen.tuxfamily.org/dox/classEigen_1_1Transform.html#a29b71eedfb78a436577878e2e5560946)